tests/core/test_actions.py

```diff
@@ -799,6 +801,135 @@ async def test_response_channel_specific(default_nlg, default_tracker, domain: D
+            - text: test
+              id: '1'
+        utter_multiple_ids:
+            - text: test
+              id: '2'
+            - text: test
+              id: '3'
+        utter_no_id:
+            - text: test
+    """
+    domain = Domain.from_yaml(domain_yaml)
+    return domain
+
+
+@pytest.fixture
+def mock_message() -> Dict[Text, Any]:
+    return {"text": "test", "response_ids": ["1"]}
+
+
+@pytest.fixture
+def mock_nlg(mock_message, monkeypatch: MonkeyPatch) -> MagicMock:
+    _mock_nlg = MagicMock()
+
+    future = asyncio.Future()
+    future.set_result(mock_message)
+
+    _mock_nlg.generate = MagicMock()
+    _mock_nlg.generate.return_value = future
+    return _mock_nlg
+
+
+async def test_action_bot_response_with_one_response_id(
+    mock_message: Dict[Text, Any],
+    mock_nlg: MagicMock,
+    default_channel,
+    default_tracker,
+    domain_with_response_ids: Domain,
+) -> None:
+    output_channel = SlackBot("DummyToken", "General")
+
+    events = await ActionBotResponse("utter_one_id").run(
+        output_channel, mock_nlg, default_tracker, domain_with_response_ids
+    )
+
+    mock_nlg.generate.assert_called_once_with(
+        "utter_one_id", default_tracker, output_channel.name(), response_ids={"1"}
+    )
+
+    assert len(events) == 1
+    assert events[0].metadata == mock_message
+
+
+async def test_action_bot_response_with_multiple_response_id(
+    mock_message: Dict[Text, Any],
+    mock_nlg: MagicMock,
+    default_channel,
+    default_tracker,
+    domain_with_response_ids: Domain,
+) -> None:
+    output_channel = SlackBot("DummyToken", "General")
+
+    events = await ActionBotResponse("utter_multiple_ids").run(
+        output_channel, mock_nlg, default_tracker, domain_with_response_ids
+    )
+
+    mock_nlg.generate.assert_called_once_with(
+        "utter_multiple_ids",
+        default_tracker,
+        output_channel.name(),
+        response_ids={"2", "3"},
+    )
+
+    assert len(events) == 1
+    assert events[0].metadata == mock_message
+
+
+async def test_action_bot_response_with_empty_response_id_set(
+    mock_message: Dict[Text, Any],
+    mock_nlg: MagicMock,
+    default_channel,
+    default_tracker,
+    domain_with_response_ids: Domain,
+) -> None:
+    output_channel = SlackBot("DummyToken", "General")
+
+    events = await ActionBotResponse("utter_no_id").run(
+        output_channel, mock_nlg, default_tracker, domain_with_response_ids
+    )
+
+    mock_nlg.generate.assert_called_once_with(
+        "utter_no_id",
+        default_tracker,
+        output_channel.name(),
+        response_ids=set(),
+    )
+
+    assert len(events) == 1
+    assert events[0].metadata == mock_message
+
+
+async def test_action_bot_response_with_non_existing_id_mapping(
+    mock_message: Dict[Text, Any],
+    mock_nlg: MagicMock,
+    default_channel,
+    default_tracker,
+    domain_with_response_ids: Domain,
+) -> None:
+    output_channel = SlackBot("DummyToken", "General")
```

Comment:
Just an observation, nothing really to action, you could also just use `CollectingOutputChannel` for unit testing.